packages/rocket-file-uploads-core/src/index.ts

```diff
@@ -1,27 +1,49 @@
 import { BoosterConfig, RocketDescriptor } from '@boostercloud/framework-types'
-import { functionID, RocketFilesParams } from '@boostercloud/rocket-file-uploads-types'
+import {
+  functionID,
+  RocketFilesConfiguration,
+  RocketFilesUserConfiguration,
+  RocketProviderPackageType,
+} from '@boostercloud/rocket-file-uploads-types'
 import { fileUploaded } from './file-uploaded'
 
 export { FileHandler } from './file-handler'
 
 export class BoosterRocketFiles {
-  public constructor(readonly config: BoosterConfig, readonly params: RocketFilesParams) {
-    config.registerRocketFunction(functionID, async (config: BoosterConfig, request: unknown) => {
-      return fileUploaded(config, request, params)
-    })
-  }
-
-  public rocketForAzure(): RocketDescriptor {
+  public rocketForAzure(config: BoosterConfig, userConfiguration: RocketFilesUserConfiguration): RocketDescriptor {
+    const configuration = BoosterRocketFiles.buildParameters(
+      userConfiguration,
+      '@boostercloud/rocket-file-uploads-azure'
+    )
```

Comment:
Then we can leave this without params and take the data from the instance:
```suggestion
  public rocketForAzure(): RocketDescriptor {
    const configuration = BoosterRocketFiles.buildParameters(
      this.params,
      '@boostercloud/rocket-file-uploads-azure'
    )
```